### PR TITLE
Fix: Update swagger tests

### PR DESCRIPTION
**Pull request by colinbruce**
## What

While working on the new CivilMeritsQuestion endpoint I realised that, because we historically used APIPie, we had a lot of test duplication.

We had request tests _and_ swagger_doc tests for all of the endpoints. This was because the request specs used to exercise APIPie and the swagger_doc tests only generated the documentation

This PR unifies the tests into a swagger test that exercises the controllers and generates the documentation

It also reduces the time taken to run the tests as, in some cases, the documentation example was built by calling a service class and then calling the endpoint that would call the service class a second time

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: spec/requests/client_involvement_types_controller_swagger_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+require "swagger_helper"
+
+RSpec.describe "ClientInvolvementTypesController", type: :request do
```
Comment (jsugarman) [inline spec/requests/client_involvement_types_controller_swagger_spec.rb:3]: For consistency with other renaming of top level describe block would this be more appropriate.
```suggestion
RSpec.describe "client_involvement_types", type: :request do
```

File: spec/requests/proceeding_types/searches_controller_swagger_spec.rb [hunk 2]
```diff
@@ -0,0 +1,223 @@
+require "swagger_helper"
+
+RSpec.describe "ProceedingTypes::SearchesController", type: :request do
```
Comment (jsugarman) [inline spec/requests/proceeding_types/searches_controller_swagger_spec.rb:3]: For consistency with other renaming of top level describe block would this be more appropriate.

```suggestion
RSpec.describe "searches", type: :request do
OR
RSpec.describe "proceeding_types/searches", type: :request do
```

File: spec/requests/proceeding_types/threshold_waivers_controller_swagger_spec.rb [hunk 3]
```diff
@@ -0,0 +1,110 @@
+require "swagger_helper"
+
+RSpec.describe "threshold_waivers", type: :request do
```
Comment (jsugarman) [inline spec/requests/proceeding_types/threshold_waivers_controller_swagger_spec.rb:3]: not sure but relative to the other threshold waivers spec perhaps
```suggestion
RSpec.describe "proceeding_types/threshold_waivers", type: :request do
```

**Review by jsugarman — COMMENTED** [review]
Looks good. Aside from a few describe naming suggestions, I have not had time to check whether all the preexisting happy and unhappy paths tests have been "converted". Is that the case? otherwise happy to approve.

**Review by kmahern — APPROVED** [review]
lgtm 👍

**Pull request merged**